cmd/clusterctl/client/upgrade.go

```diff
@@ -255,3 +267,34 @@ func parseUpgradeItem(ref string, providerType clusterctlv1.ProviderType) (*clus
 		NextVersion: version,
 	}, nil
 }
+
+func parseUpgradeItemWithoutNamespace(clusterClient cluster.Client, ref string, providerType clusterctlv1.ProviderType) (*cluster.UpgradeItem, error) {
+	if !strings.Contains(ref, ":") {
+		return nil, errors.Errorf("invalid provider name %q. Provider name should be in the form namespace/name[:version] and namespace cannot be empty", ref)
+	}
+
+	name, version, err := parseProviderName(ref)
+	if err != nil {
+		return nil, errors.Wrapf(err, "invalid provider name %q. Provider name should be in the form namespace/name[:version] and the namespace should be valid", ref)
```

Comment:
nit
```suggestion
		return nil, errors.Wrapf(err, "invalid provider name %q. Provider name should be in the form namespace/name[:version] and the namespace should be valid", ref)
```

Reply:
Thanks! I broke out the error message since a similar message is used in multiple places and needed updating everywhere.